lib/colors.js

```diff
@@ -107,12 +107,26 @@ function applyStyle() {
   var nestedStyles = this._styles;
 
   var i = nestedStyles.length;
+
+  var rebuiltString = '';
+
   while (i--) {
     var code = ansiStyles[nestedStyles[i]];
-    str = code.open + str.replace(code.closeRe, code.open) + code.close;
+
+    // multiline handling functionality
+    var lines = str.split(/[\r\n]+/)
```

Comment:
Could this not be done with a one-line replace all, i.e. replace all `'\n'` with `code.close + '\n' + code.open` ?